include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -144,7 +151,7 @@ class MjbotsControlLoop : public AbstractRealtimeObject {
   void SetupOptions(const ControlLoopOptions &options);
 
   std::shared_ptr<RobotClass> robot_;     /// ptr to the robot object
-  std::shared_ptr<kodlab::mjbots::MjbotsHardwareInterface> mjbots_interface_;   ///ptr to mjbots_interface object, if unique causes issues, also should be initialized inside thread
+  std::shared_ptr<INTERFACE_TYPE> robot_interface_;
```

Comment:
```suggestion
  std::shared_ptr<kodlab::RobotInterface> robot_interface_;
```

Reply:
Raise compiler error because HardwareInterface and SimulationInterface have some extra functions, which should not live in RobotInterface (such as getJointModes), being called by control loop or examples.

Put a command instead. "//INTERFACE_TYPE is a child class of RobotInterface"